### PR TITLE
CORE-10127: Early fail of re-registration in static network

### PR DESCRIPTION
**Pull request by yift-r3**

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]
LGTM. Is it worth checking if the resilience tests have stopped attempting to repeatedly hit the register endpoint before merging? Not sure if the QA tests or stability tests are trying to do this. I think not.

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt [hunk 1]
```diff
@@ -177,6 +180,14 @@ class StaticMemberRegistrationService @Activate constructor(
                 ex,
             )
         }
+        val membershipGroupReader = membershipGroupReaderProvider.getGroupReader(member)
+        val alreadyRegisterMember = membershipGroupReader.lookup(member.x500Name)
+        if (alreadyRegisterMember?.isActive == true) {
+            throw InvalidMembershipRegistrationException(
+                "The member ${member.x500Name} had been register successfully in the group ${member.groupId}. " +
```
Comment (dimosr) [inline components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt]:
```suggestion
                "The member ${member.x500Name} had been registered successfully in the group ${member.groupId}. " +
```

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt [hunk 2]
```diff
@@ -177,6 +180,14 @@ class StaticMemberRegistrationService @Activate constructor(
                 ex,
             )
         }
+        val membershipGroupReader = membershipGroupReaderProvider.getGroupReader(member)
+        val alreadyRegisterMember = membershipGroupReader.lookup(member.x500Name)
```
Comment (dimosr) [inline components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/staticnetwork/StaticMemberRegistrationService.kt]:
```suggestion
        val alreadyRegisteredMember = membershipGroupReader.lookup(member.x500Name)
```

**Review by dimosr — COMMENTED** [review]
LGTM - just a couple of very minor comments.

Also, agree with Charlie please check with Juan whether adjustments are needed for the resiliency tests and announce the change in behaviour in the Slack channels when it's merged.

**Review by dimosr — APPROVED** [review]

**Pull request merged**